src/riak_kv_w1c_worker.erl

```diff
@@ -149,12 +155,37 @@ ts_batch_put(RObjs, W, PW, Bucket, NVal, EncodeFn, Preflist) ->
 
     gen_server:cast(
       Worker,
-      {batch_put, EncodedVals, ReqId, Preflist,
+      {batch_put, Bucket, BucketProps, EncodedVals, ReqId, DocIdx, Preflist,
        #rec{w=W, pw=PW, n_val=NVal, from=self(),
             start_ts=StartTS,
             size=Size}}),
     {ok, {ReqId, Worker}}.
 
+%% Primarily for repl support. Put a list of objects represented as
+%% tuple({Bucket, LocalKey}, msgpack-encoded riak object) into the
+%% correct partition
+ts_batch_put_encoded([{{Bucket, _LK}, _RObj0}|_Rest]=RObjs, DocIdx) ->
```

Comment:
The funcs ts_batch_put_encoded and ts_batch_put contain a lot of common code which could be refactored out - both funs being a 'prep' stage for a do_put function